In [2]:
import pandas as pd
import duckdb
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

In [ ]:
df = pd.read_csv('Flights_2018_Modified (1).csv')
unique = df['Origin'].unique()
for airport in unique:
    print(airport)

In [8]:
len(unique)

370

In [20]:
airports = pd.read_csv("airports.csv")
airport_list = airports['IATA']
uniqueSet = set(unique)
airport_list_set = set(airport_list)

notInList = uniqueSet - airport_list_set
# notInList = airport_list_set - uniqueSet
# len(notInList)
for airport in notInList:
    print(airport)

PSM
USA
BLV
YKM
AZA
HGR
IPT
LNY
CGI
LBE
CKB
ECP
PQI
PGV
SCK
PBG
OWB
PVU
ROP
JHM
OGD
LCK
EAR
SLN
UIN
HVN
PUW
PRC
LBL
EAT
ALW
MKK
HHH
BKG
STS
OGS
SFB
BFF
PGD
YNG
SWO
DUT
LBF
IAG
SBY


In [6]:
#testing nunique
x = {'col1': ['IAD', 'IAD', 'IAD', 'SYR', 'SYR', 'SYR']}
test = pd.DataFrame(data=x)
test['col1'].nunique()

2

It is not realistic to be able to scrape 2018 weather data for 370 airports. Each location takes approximately 1.2 hours so it would take around 444 hours to scrape all data.